sklearn/inspection/tests/test_permutation_importance.py

```diff
@@ -351,3 +351,55 @@ def test_permutation_importance_large_memmaped_data(input_type):
     # permutating feature should not change the predictions
     expected_importances = np.zeros((n_features, n_repeats))
     assert_allclose(expected_importances, r.importances)
+
+
+def test_permutation_importance_sample_weight():
+    # Creating data with 2 features and 1000 samples, where the target
+    # variable is a linear combination of the two features, such that
+    # in half of the samples the impact of feature 1 is twice the impact of
+    # feature 2, and vice versa on the other half of the samples.
+    rng = np.random.RandomState(1)
+    n_samples = 1000
+    n_features = 2
+    n_half_samples = n_samples // 2
+    x = rng.normal(0.0, 0.001, (n_samples, n_features))
+    y = np.zeros(n_samples)
+    y[:n_half_samples] = 2 * x[:n_half_samples, 0] + x[:n_half_samples, 1]
+    y[n_half_samples:] = x[n_half_samples:, 0] + 2 * x[n_half_samples:, 1]
+
+    # Fitting linear regression with perfect prediction
+    lr = LinearRegression(fit_intercept=False)
+    lr.fit(x, y)
+
+    # When all samples are weighted with the same weights, the ratio of
+    # the two features importance should equal to 1 on expectation (when using
+    # mean absolutes error as the loss function).
+    pi = permutation_importance(lr, x, y, random_state=1,
+                                scoring='neg_mean_absolute_error',
+                                n_repeats=1000)
```

Comment:
Can we lower the `n_repeats` and relax the `pytest.approx` a little?

Lowering `n_repeats` would help with the runtime of this test.

Reply:
Changed to 200